tests/SassSpecTest.php

```diff
@@ -187,28 +188,30 @@ public function testTests($name, $input, $output)
         // normalize css for comparison purpose
         $css = $this->normalizeCssOutput($css);
 
-        // ignore tests expecting an error for the moment
-        if (! strlen($error)) {
-            $fp_err_stream = fopen("php://memory", 'r+');
-            static::$scss->setErrorOuput($fp_err_stream);
-
-            // this test needs @import of includes files, build a dir with files and set the ImportPaths
-            if ($includes) {
-                $basedir = sys_get_temp_dir() . '/sass-spec/' . preg_replace(",^\d+/\d+\.\s*,", "", $name);
-
-                foreach ($includes as $f => $c) {
-                    $f = $basedir . '/' . $f;
+        // this test needs @import of includes files, build a dir with files and set the ImportPaths
+        if ($includes) {
+            $basedir = sys_get_temp_dir() . '/sass-spec/' . preg_replace(",^\d+/\d+\.\s*,", "", $name);
 
-                    if (! is_dir(dirname($f))) {
-                        passthru("mkdir -p " . dirname($f));
-                    }
+            foreach ($includes as $f => $c) {
+                $f = $basedir . '/' . $f;
 
-                    file_put_contents($f, $c);
+                if (! is_dir(dirname($f))) {
+                    passthru("mkdir -p " . dirname($f));
                 }
 
-                static::$scss->setImportPaths([$basedir]);
+                file_put_contents($f, $c);
             }
 
+            static::$scss->setImportPaths([$basedir]);
+        } else {
+            static::$scss->setImportPaths([]);
```

Comment:
this resetting ensures that we reset the state of the shared compiler even if a previous test failed to reset it properly (due to an error for instance), which improves test isolation.

I'm actually considering instantiating a separate compiler per test instead to improve isolation, but that's out of the scope of this PR.

Reply:
I used to have compilation memory issue when wanting to run tests suite over the complete Sass-spec and I was finaly able to do that only with a static instance of the compiler.

But could also be because there was a lot of errors&fails at this time (and memory leaks?), maybe now it could work, so go for a try if you want!

Reply:
I tried it, and PHPUnit reports exactly the same memory usage